src/math_util.c

```diff
@@ -708,6 +707,11 @@ DOUBLE_angle(char **args, intp *dimensions, intp *steps, void *NPY_UNUSED(func))
         dot_qd(B, X, &diff);
         dot_qd(ABX, BCX, &inner);
 
+        if (fabs(inner.x[0]) == 1.0 & fabs(inner.x[1]) < 1e-60) {
```

Comment:
Does this _need to use_ `fabs` or maybe a slightly less restrictive check would be OK?
```suggestion
        if ((inner.x[0] == 1.0 & inner.x[1]) < 1e-60 & inner.x[1]) > 0) | (inner.x[0] == -1.0 & inner.x[1]) > -1e-60 & inner.x[1]) < 0)) {
```

Reply:
I pulled it out of the air after seeing the typical nonzero values that occurred when the one value changed. Ultimately, we don't know the fundamental source of the bug. If it resurfaces with larger values we can up the threshold.

Reply:
Is there a value in adding a new test in this PR or is this already tested in the current test suite?

Reply:
I will add a comment about the threshold and where it came from

Reply:
And actually, since we know `|dot(unit vectors)|<=1`, maybe just do something like this:

```python
if ((inner.x[0] == 1.0 & inner.x[1]) > 0) | (inner.x[0] == -1.0 & inner.x[1]) < 0)) {
```

Reply:
And still that bias against `fabs()`